api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculator.java

```diff
@@ -43,15 +57,53 @@ private void validateArguments(Integer paymentCycleDurationInDays, Integer numbe
         }
     }
 
-    public PaymentCycleVoucherEntitlement calculateCycleVoucherEntitlement(Claimant claimant, List<LocalDate> dateOfBirthOfChildren) {
+    /**
+     * Calculates the total voucher entitlement for a payment cycle when there is no previous voucher entitlement.
+     * @param expectedDueDate expected due date
+     * @param dateOfBirthOfChildren the date of birth of the claimant's children
+     * @return the payment cycle voucher entitlement calculated for the claimant
+     */
+    public PaymentCycleVoucherEntitlement calculateEntitlement(Optional<LocalDate> expectedDueDate, List<LocalDate> dateOfBirthOfChildren) {
         List<LocalDate> entitlementDates = getEntitlementDates();
         List<VoucherEntitlement> voucherEntitlements = entitlementDates.stream()
-                .map(date -> entitlementCalculator.calculateVoucherEntitlement(claimant, dateOfBirthOfChildren, date))
+                .map(date -> entitlementCalculator.calculateVoucherEntitlement(expectedDueDate, dateOfBirthOfChildren, date))
                 .collect(Collectors.toList());
 
         return new PaymentCycleVoucherEntitlement(voucherEntitlements);
     }
 
+    /**
+     * Calculates the total voucher entitlement for a payment cycle.
+     * @param expectedDueDate expected due date
+     * @param dateOfBirthOfChildren the date of birth of the claimant's children
+     * @param previousVoucherEntitlement voucher entitlement from last payment cycle
+     * @return the payment cycle voucher entitlement calculated for the claimant
+     */
+    public PaymentCycleVoucherEntitlement calculateEntitlement(Optional<LocalDate> expectedDueDate,
+                                                               List<LocalDate> dateOfBirthOfChildren,
+                                                               PaymentCycleVoucherEntitlement previousVoucherEntitlement) {
+        if (newChildrenMatchedToExpectedDeliveryDate(expectedDueDate, dateOfBirthOfChildren, previousVoucherEntitlement)) {
+            // ignore expected due date as we've determined that the pregnancy has happened
+            return calculateEntitlement(Optional.empty(), dateOfBirthOfChildren);
+        }
+
+        return calculateEntitlement(expectedDueDate, dateOfBirthOfChildren);
+    }
+
+    // A child is considered a result of expected due date if their date of birth falls within a range of that date
+    private boolean newChildrenMatchedToExpectedDeliveryDate(Optional<LocalDate> expectedDueDate,
+                                                             List<LocalDate> dateOfBirthOfChildren,
+                                                             PaymentCycleVoucherEntitlement previousVoucherEntitlement) {
+        if (previousVoucherEntitlement.getVouchersForPregnancy() == 0 || expectedDueDate.isEmpty()) {
+            return false;
+        }
+
+        LocalDate startDate = expectedDueDate.get().minusWeeks(weeksBeforeDueDate);
```

Comment:
How about we create a method to encapsulate the period matching, so this becomes:
```
        return dateOfBirthOfChildren.stream()
                .anyMatch(date -> isWithPregnancyMatchPeriod(expectedDueDate, date));
    }

    private boolean isWithPregnancyMatchPeriod(Optional<LocalDate> expectedDueDate, LocalDate date) {
        LocalDate startDate = expectedDueDate.get().minusWeeks(weeksBeforeDueDate);
        LocalDate endDate = expectedDueDate.get().plusWeeks(weeksAfterDueDate);
        return !date.isBefore(startDate) && !date.isAfter(endDate);
    }
```

Reply:
done.

Reply:
Sorry, my mistake here, the method should be called `isWithinPregnancyMatchPeriod`.

Reply:
I should have noticed, updated.